File: frontend/src/types/hardware.ts
export interface CPUInfo {
  name: string
  cores_physical: number
  cores_logical: number
  frequency_mhz: number
  usage_percent: number
  architecture: string
}

export interface RAMInfo {
  total_gb: number
  available_gb: number
  used_gb: number
  usage_percent: number
}

export interface GPUInfo {
  index: number
  name: string
  vram_total_gb: number
  vram_free_gb: number
  vram_used_gb: number
  usage_percent: number
  temperature_c: number | null
  cuda_version: string | null
}

export interface StorageInfo {
  total_gb: number
  free_gb: number
  used_gb: number
  usage_percent: number
}

export interface HardwareProfile {
  cpu: CPUInfo
  ram: RAMInfo
  gpus: GPUInfo[]
  storage: StorageInfo
  os_name: string
  os_version: string
  has_gpu: boolean
  is_cuda_available: boolean
  total_vram_gb: number
  readiness_score: number
  readiness_status: string
}

File: llmstore/backend/app/models/hardware.py
from pydantic import BaseModel
from typing import List, Optional

class CPUInfo(BaseModel):
    name: str
    cores_physical: int
    cores_logical: int
    frequency_mhz: float
    usage_percent: float
    architecture: str

class RAMInfo(BaseModel):
    total_gb: float
    available_gb: float
    used_gb: float
    usage_percent: float

class GPUInfo(BaseModel):
    index: int
    name: str
    vram_total_gb: float
    vram_free_gb: float
    vram_used_gb: float
    usage_percent: float
    temperature_c: Optional[float] = None
    cuda_version: Optional[str] = None

class StorageInfo(BaseModel):
    total_gb: float
    free_gb: float
    used_gb: float
    usage_percent: float

class HardwareProfile(BaseModel):
    cpu: CPUInfo
    ram: RAMInfo
    gpus: List[GPUInfo]
    storage: StorageInfo
    os_name: str
    os_version: str
    has_gpu: bool
    is_cuda_available: bool
    total_vram_gb: float
    readiness_score: float
    readiness_status: str
